examples/scala/src/main/scala/StandaloneExample.scala

```diff
@@ -0,0 +1,128 @@
+     * PING using the provided <code>GlideClient</code>.
+     *
+     * @param client An instance of <code>GlideClient</code>.
+     */
+    private def appLogic(client: GlideClient): Future[Unit] = {
+        for {
+            // Send SET and GET
+            setResponse <- client.set("foo", "bar").asScala
+            _ = Logger.log(Logger.Level.INFO, "app", s"Set response is $setResponse")
+
+            getResponse <- client.get("foo").asScala
+            _ = Logger.log(Logger.Level.INFO, "app", s"Get response is $getResponse")
+
+            // Send PING to the primary node
+            pong <- client.ping().asScala
+            _ = Logger.log(Logger.Level.INFO, "app", s"Ping response is $pong")
+        } yield ()
+    }
+
+    /**
+     * Executes the application logic with exception handling.
+     */
+    private def execAppLogic(): Future[Unit] = {
+        def loop(): Future[Unit] = {
+            createClient().flatMap(client => appLogic(client).andThen {
+                case _ => Try(client.close()) match {
+                    case Failure(e) =>
+                        Logger.log(
+                            Logger.Level.WARN,
+                            "glide",
+                            s"Encountered an error while closing the client: ${e.getCause}"
+                        )
+                    case _ => ()
+                }
+            }).recoverWith {
+                case e: CancellationException =>
+                    Logger.log(Logger.Level.ERROR, "glide", s"Request cancelled: ${e.getMessage}")
+                    Future.failed(e)
+                case e: Exception => e.getCause match {
+                    case e: ClosingException if e.getMessage.contains("NOAUTH") =>
+                        // If the error message contains "NOAUTH", raise the exception
+                        // because it indicates a critical authentication issue.
+                        Logger.log(Logger.Level.ERROR, "glide", s"Authentication error encountered: ${e.getMessage}")
+                        Future.failed(e)
+                    case e: ClosingException =>
+                        Logger.log(Logger.Level.WARN, "glide", s"Client has closed and needs to be re-created: ${e.getMessage}")
+                        loop()
+                    case e: TimeoutException =>
+                        // A request timed out. You may choose to retry the execution based on your application's logic
+                        Logger.log(Logger.Level.ERROR, "glide", s"Timeout encountered: ${e.getMessage}")
+                        Future.failed(e)
+                    case e: ConnectionException =>
+                        // The client wasn't able to reestablish the connection within the given retries
+                        Logger.log(Logger.Level.ERROR, "glide", s"Connection error encountered: ${e.getMessage}")
+                        Future.failed(e)
+                    case e: ExecAbortException =>
```

Comment:
dont catch ExecAbortException. it should be catched as a part of  the general Execution error.